pkg/autoupdate/findnew.go

```diff
@@ -266,6 +263,41 @@ func getUpdateDir(fullBinaryPath string) string {
 	return fmt.Sprintf("%s%s", components[0], updateDirSuffix)
 }
 
+// Find the possible updates. filepath.Glob returns a list of things
+// that match the requested pattern. We sort the list to ensure that
+// we can tell which ones are earlier or later (remember, these are
+// timestamps).
+func getPossibleUpdates(updateDir, binaryName string) ([]string, error) {
+	// If this is launcher running on macOS, then we should have app bundles available instead --
+	// check for those first.
+	if runtime.GOOS == "darwin" {
+		binarySuffix := filepath.Join("Contents", "MacOS", binaryName)
+		fileGlob := filepath.Join(updateDir, "*", "*.app", binarySuffix)
+		possibleUpdates, err := filepath.Glob(fileGlob)
+
+		if err == nil && len(possibleUpdates) > 0 {
+			appBundleNames := make([]string, len(possibleUpdates))
+			for i, binaryPath := range possibleUpdates {
+				appBundleNames[i] = strings.TrimSuffix(binaryPath, "/"+binarySuffix)
+			}
```

Comment:
I'm not sure I understand _why_ this strips off the `binarySuffix`. The caller simply re-adds it, so why not return the binary directy?

Reply:
It makes it a little tidier for the caller to grab the timestamp out of the path to perform timestamp comparisons (can always be `updateDownloadTime := filepath.Base(basedir)`).

Reply:
Oh, it's also required so that the cleanup of old/corrupt updates works without any modifications needed -- that was the primary reason I ended up doing it like this.

Reply:
_Ah._ So it's this way to avoider a larger PR. While there are probably clean ways to handle it, I think it's probably reasonable to be narrow here -- but comment about the reasoning please. 
```suggestion
			for i, binaryPath := range possibleUpdates {
			        // We trim and re-add the suffix for compatibility with the prior setup. Things like cleanup and
			        // timezone comparisons work. 
				appBundleNames[i] = strings.TrimSuffix(binaryPath, "/"+binarySuffix)
			}
```

Reply:
Added a comment both here and where we're re-adding the suffix 🙂